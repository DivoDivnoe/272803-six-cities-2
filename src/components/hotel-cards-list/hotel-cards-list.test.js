import React from 'react';
import renderer from 'react-test-renderer';
import HotelCardsList from './hotel-cards-list.jsx';

jest.mock(`../hotel-card/hotel-card.jsx`, () => jest.fn().mockReturnValue(null));

describe(`HotelCardsList component`, () => {
  it(`is rendered correctly`, () => {
    const classNames = [``];

    const offers = [
      {
        id: 1,
        city: {
          name: `somecity`,
          location: {
            latitude: 0,
            longitude: 0,
            zoom: 1
          }
        },
        type: `apartment`,
        price: 0,
        rating: 0,
        previewImage: `some/src`,
        images: [`some/src`],
        title: `some title`,
        goods: [`first`, `second`],
        bedrooms: 0,
        maxAdults: 0,
        description: `some desc.`,
        isPremium: true,
        isFavorite: false,
        host: {
          id: 1,
          isPro: true,
          name: `Angelina`,
          avatarUrl: `some/src`
        },
        location: {
          latitude: 0,
          longitude: 0,
          zoom: 1
        }
      }
    ];
    const onActiveHotel = jest.fn();
    const onDisactiveHotel = jest.fn();

    const tree = renderer
      .create(
          <HotelCardsList
            offers={offers}
            classNames={classNames}
            onActiveHotel={onActiveHotel}
            onDisactiveHotel={onDisactiveHotel}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
