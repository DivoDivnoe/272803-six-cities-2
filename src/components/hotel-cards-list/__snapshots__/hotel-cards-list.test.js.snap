// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelCardsList component is rendered correctly 1`] = `
<div
  className=""
/>
`;
